### PR TITLE
added SuperServer SSG-640P-E1CR24H

### PR DESCRIPTION
**Pull request by TaLoN1x**
added SuperServer SSG-640P-E1CR24H

### CODE REVIEW

File: device-types/Supermicro/SSG-6028R-E1CR16T.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+console-ports:
+  - name: Serial
+    type: de-9
+power-ports:
+  - name: PSU-1
+    type: iec-60320-c14
+    maximum_draw: 1000
+  - name: PSU-2
+    type: iec-60320-c14
+    maximum_draw: 1000
```
Comment (danner26) [inline device-types/Supermicro/SSG-6028R-E1CR16T.yml:21]: Are these PSU's modular? If so, they should be modelled under `module-bays` and the associated PSU modules should be added to the module-types folder if they do not already exist

File: device-types/Supermicro/SSG-6029P-E1CR12T.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+console-ports:
+  - name: Serial
+    type: de-9
+power-ports:
+  - name: PSU-1
+    type: iec-60320-c14
+    maximum_draw: 1200
+  - name: PSU-2
+    type: iec-60320-c14
+    maximum_draw: 1200
```
Comment (danner26) [inline device-types/Supermicro/SSG-6029P-E1CR12T.yml:21]: Are these PSU's modular? If so, they should be modelled under `module-bays` and the associated PSU modules should be added to the module-types folder if they do not already exist

File: device-types/Supermicro/SSG-640P-E1CR24H.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+console-ports:
+  - name: Serial
+    type: de-9
+power-ports:
+  - name: PSU-1
+    type: iec-60320-c14
+    maximum_draw: 1200
+  - name: PSU-2
+    type: iec-60320-c14
+    maximum_draw: 1200
```
Comment (danner26) [inline device-types/Supermicro/SSG-640P-E1CR24H.yml:21]: Are these PSU's modular? If so, they should be modelled under `module-bays` and the associated PSU modules should be added to the module-types folder if they do not already exist

File: device-types/Supermicro/SSG-640P-E1CR24H.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    position: PCIe-1
+  - name: PCIe-2 LP x16
+    position: PCIe-2
+  - name: PCIe-3 LP x16
+    position: PCIe-3
+  - name: PCIe-4 LP x8
+    position: PCIe-4
+  - name: PCIe-5 NVME
+    position: NVME
+    comment: M-Key M.2
```
Comment (danner26) [inline device-types/Supermicro/SSG-640P-E1CR24H.yml:37]: Right now, storage should not be modelled in this repo

File: device-types/Supermicro/SSG-6029P-E1CR12T.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    position: PCIe-3
+  - name: PCIe-4 LP x8
+    position: PCIe-4
+  - name: PCIe-5 LP x8
+    position: PCIe-5
+  - name: PCIe-6 LP x8
+    position: PCIe-6
+  - name: PCIe-7 LP x8
+    position: PCIe-7
+    comment: M-Key M.2
```
Comment (danner26) [inline device-types/Supermicro/SSG-6029P-E1CR12T.yml:37]: Right now, storage should not be modelled in this repo

File: device-types/Supermicro/SSG-6028R-E1CR16T.yml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    position: PCIe-3
+  - name: PCIe-4 LP x8
+    position: PCIe-4
+  - name: PCIe-5 LP x8
+    position: PCIe-5
+  - name: PCIe-6 LP x8
+    position: PCIe-6
+  - name: PCIe-7 LP x8
+    position: PCIe-7
+    comment: M-Key M.2
```
Comment (danner26) [inline device-types/Supermicro/SSG-6028R-E1CR16T.yml:37]: Right now, storage should not be modelled in this repo

**Review by danner26 — CHANGES_REQUESTED** [review]

**Review by harryajc — APPROVED** [review]

**Pull request merged**